### ISSUE TITLE
Question on whether to support Baichuan2

### ISSUE THREAD

**Issue by venxzw** [vllm-project/vllm#1459]
Excuse me, everyone. I have modified the relevant files based on the solutions in the community, but using the API_ The following error occurred during client.py's inference:
![1b67679c50d282d02e40626260b9742](https://github.com/vllm-project/vllm/assets/138442712/1672c44a-66cd-4a44-8619-6640467d2a07)
Then I added a judgment to the file as follows:
![8224a838f0eb05562921befdd5d2a77](https://github.com/vllm-project/vllm/assets/138442712/2e3e4e12-8b9f-4b29-9ac0-a9d0cc1f587d)
The inference result was not output:
![c9fa7938a783880585a5c3b476511e5](https://github.com/vllm-project/vllm/assets/138442712/6ce5b235-1e63-4a3b-b0d8-80cfccaf4161)
So I would like to ask if the framework's support for Baihuan2 has not yet been launched.

**Comment by AmazeQiu** [comment]
Not yet, I modified it to support baichuan2 by myself.  Actually baichuan and baichuan2 is really closed. The  difference of the structure is just the NormHead layer.
vllm/model_executor/models/baichuan.py
`class NormHead(ColumnParallelLinear):
    def __init__(self, hidden_size, vocab_size, bias=False):
        super().__init__(hidden_size, vocab_size, bias=False, gather_output=False)
        self.first_flag = True
    
    def get_weight(self):
        if self.first_flag:
            self.first_flag = False
            self.weight = nn.Parameter(nn.functional.normalize(self.weight))
        return self.weight

    def forward(self, hidden_states):
        if self.first_flag:
            self.first_flag = False
            self.weight = nn.Parameter(nn.functional.normalize(self.weight))
        return ColumnParallelLinear.forward(self, hidden_states)`

**Comment by venxzw** [comment]
> Not yet, I modified it to support baichuan2 by myself. Actually baichuan and baichuan2 is really closed. The difference of the structure is just the NormHead layer.

Thank you for your reply. Have you successfully accelerated Baihuan2 after making your own modifications? If so, could you please tell me which parts need to be modified? Thank you very much.

**Comment by kar9999** [comment]
> Not yet, I modified it to support baichuan2 by myself. Actually baichuan and baichuan2 is really closed. The difference of the structure is just the NormHead layer.

May I ask if the answer of vllm Baichuan is the same as that of huggingface, and if the quality of the answer has decreased？

**Comment by AmazeQiu** [comment]
> > Not yet, I modified it to support baichuan2 by myself. Actually baichuan and baichuan2 is really closed. The difference of the structure is just the NormHead layer.
> 
> Thank you for your reply. Have you successfully accelerated Baihuan2 after making your own modifications? If so, could you please tell me which parts need to be modified? Thank you very much.

Try the forked version I published: #1533

**Comment by AmazeQiu** [comment]
> > Not yet, I modified it to support baichuan2 by myself. Actually baichuan and baichuan2 is really closed. The difference of the structure is just the NormHead layer.
> 
> May I ask if the answer of vllm Baichuan is the same as that of huggingface, and if the quality of the answer has decreased？

For the lastest version, they are almost the same. Maybe slightly different, but the performance on our own test set didn't decrease.

**Comment by kar9999** [comment]
> > > 还没有，我自己修改支持了baichuan2。其实百川和百川2真的是关闭了。结构的区别只是NormHead层。
> > 
> > 
> > 请问vllm百川的答案和huggingface的答案是否一样，答案质量是否有所下降？
> 
> 对于最新版本，它们几乎是相同的。也许略有不同，但我们自己的测试集上的性能并没有下降。

![ans](https://github.com/vllm-project/vllm/assets/77961128/ca442af9-8c6d-487c-b3c0-9a0b8387f14b)
![image](https://github.com/vllm-project/vllm/assets/77961128/b9a8b388-37ea-4dc3-ae5e-9d727caadba6)
Their answers of baichuan1 are almost different.

**Comment by kar9999** [comment]
> > > 还有，我自己支持了baichuan2。其实百川和百川2真的是关闭了。结构的区别只是Norm修改Head层。
> > 
> > 
> > 请问百川的答案和huggingface的答案是否一样，答案质量是否有所下降？
> 
> 对于最新版本，它们几乎是相同的。也许不同，但我们自己的测试集上的性能并没有下降。
huggingface -baichuan1
![image](https://github.com/vllm-project/vllm/assets/77961128/54974b26-679f-41ca-80d7-26649b3aef18)
![ans](https://github.com/vllm-project/vllm/assets/77961128/f68771c3-5a8f-4be2-922c-b3954c81ee5a)
vllm-baichaun1
![image](https://github.com/vllm-project/vllm/assets/77961128/9bcbf136-9872-4cbf-b680-4e9fc429f06f)
![image](https://github.com/vllm-project/vllm/assets/77961128/69cdad42-63e9-4968-9436-b57fa22be07e)

their answears are different.

**Comment by venxzw** [comment]
> kar9999

Thank you very much for your guidance

**Comment by venxzw** [comment]
> > > Not yet, I modified it to support baichuan2 by myself. Actually baichuan and baichuan2 is really closed. The difference of the structure is just the NormHead layer.
> > 
> > 
> > Thank you for your reply. Have you successfully accelerated Baihuan2 after making your own modifications? If so, could you please tell me which parts need to be modified? Thank you very much.
 > Try the forked version I published: #1533

**Comment by venxzw** [comment]
> > > Not yet, I modified it to support baichuan2 by myself. Actually baichuan and baichuan2 is really closed. The difference of the structure is just the NormHead layer.
> > 
> > 
> > May I ask if the answer of vllm Baichuan is the same as that of huggingface, and if the quality of the answer has decreased？
> 
> For the lastest version, they are almost the same. Maybe slightly different, but the performance on our own test set didn't decrease.

Thank you very much for your guidance

**Comment by ouwenjie03** [comment]
> > > > 还有，我自己支持了baichuan2。其实百川和百川2真的是关闭了。结构的区别只是Norm修改Head层。
> > > 
> > > 
> > > 请问百川的答案和huggingface的答案是否一样，答案质量是否有所下降？
> > 
> > 
> > 对于最新版本，它们几乎是相同的。也许不同，但我们自己的测试集上的性能并没有下降。
> > huggingface -baichuan1
> > ![image](https://user-images.githubusercontent.com/77961128/279618425-54974b26-679f-41ca-80d7-26649b3aef18.png)
> > ![ans](https://user-images.githubusercontent.com/77961128/279617893-f68771c3-5a8f-4be2-922c-b3954c81ee5a.png)
> > vllm-baichaun1
> > ![image](https://user-images.githubusercontent.com/77961128/279618264-9bcbf136-9872-4cbf-b680-4e9fc429f06f.png)
> > ![image](https://user-images.githubusercontent.com/77961128/279618069-69cdad42-63e9-4968-9436-b57fa22be07e.png)
> 
> their answears are different.

It looks like you had a similar problem as mine #1561 . May I ask how you solved it?

**Comment by firebook** [comment]
I think this issue should be closed，the latest vllm version support baichuan2，related issue：#1876 #1874
